File: cracking_the_coding_interview/1.1isUniqueChars.py
def isUniqueChars(str):
    if len(str) > 256: 
        return False
    
    strCheck = [False] * 256
    for ch in str:
        if strCheck[ord(ch)]: 
            return False
        else:
            strCheck[ord(ch)] = True
            
    return True
        
str = "aabcd"
res = isUniqueChars(str)    
print res

File: cracking_the_coding_interview/1.1isUniqueChars.php
<?php

  $res = isUniqueChars("abcde");
  
  echo "result:".var_export($res, 1); ;

  function isUniqueChars($str){
    if (strlen($str) > 256) return false; 
    $check = array_fill(0, 255, false);
    $strArray = str_split($str);
    foreach ($strArray as $ch) 
    {
      $idx = ord($ch);
      if ($check[$idx] == true) 
      {
        return false;
      }
      else 
      {
        $check[$idx] = true;
      } 
    }
    return true;
  }

?>